### PR TITLE
feature/DEV-77_v0.2.1-Hook-up-Onboarding-API

### PR DESCRIPTION
**Pull request by thedigitaldesign**
- Switched over Profile to new API …
- Ran through heavy debugging for a multiple loading of Terms, found the issue and commented out registerMachine()

### CODE REVIEW

File: packages/web-app/src/modules/profile/models/Profile.ts [hunk 1]
```diff
@@ -2,12 +2,9 @@ export interface Profile {
   id: string
   username: string
   email: string
-  isNewUser: boolean
-  termsOfService?: string
-  whatsNewVersion?: string
-  referred?: ReferredStatus
-  trackUsageVersion?: string
-  tutorialComplete: boolean
+  lastAcceptedTermsOfService: string | null | undefined
```
Comment (dsarfati) [inline packages/web-app/src/modules/profile/models/Profile.ts]: Why include `null`?

File: packages/web-app/src/modules/profile/models/ProfileResource.ts [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-  termsOfService: string
-  trackUsageVersion: string
-  tutorialComplete: number
-  whatsNewVersion: string
-  profileData: {
-    email: string
-    name: string
-    nickname: string
+  id: string
+  auth0Id: string
```
Comment (dsarfati) [inline packages/web-app/src/modules/profile/models/ProfileResource.ts]: Is the `auth0Id` required here? I don't return this value now

File: packages/web-app/src/modules/profile/models/ProfileResource.ts [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
+  created: string
+  username: string
+  email: string
+  onboarding: {
```
Comment (dsarfati) [inline packages/web-app/src/modules/profile/models/ProfileResource.ts]: The API resource does not have these values under an `onboarding` property. Just wanna make sure that we are in sync here.

File: packages/web-app/src/modules/profile/models/Profile.ts (hunk 1 above)
Comment (thedigitaldesign) [reply packages/web-app/src/modules/profile/models/Profile.ts]: Removed. It was leftover from when I was unclear on how things were setup.

File: packages/web-app/src/modules/profile/models/ProfileResource.ts (hunk 2 above)
Comment (thedigitaldesign) [reply packages/web-app/src/modules/profile/models/ProfileResource.ts]: Removed

File: packages/web-app/src/modules/profile/models/ProfileResource.ts (hunk 3 above)
Comment (thedigitaldesign) [reply packages/web-app/src/modules/profile/models/ProfileResource.ts]: Flattened

**Pull request closed without merging**